Example/Core/Tests/FIRAppTest.m

```diff
@@ -464,35 +459,19 @@ - (void)testAppIDFingerprintInvalid {
   XCTAssertFalse([FIRApp validateAppIDFingerprint:kGoodVersionV1 withVersion:kGoodVersionV1]);
   // The version is the entire app ID.
   XCTAssertFalse([FIRApp validateAppIDFingerprint:kGoodAppIDV1 withVersion:kGoodAppIDV1]);
-
-  // Versions digits that may make a partial match.
-  XCTAssertFalse([FIRApp validateAppIDFingerprint:@"01:1337:ios:deadbeef"
```

Comment:
@paulb777  The method [FIRApp validateAppIDFingerprint:] is a private one, so it is not called from outside with an arbitrary AppID. The method has a single responsibility - validate the fingerprint part for an AppID version. We validate the format and all other components before calling this method. Having this in mind, we can remove all unrelated validations (as well as the tests for them). The same checks now are present only for `testAppIDFormatInvalid`

Another point - from my experience, it is better not to test private methods, but test only public interface of a class. Then it is easier to keep tests more expressive. If we tested only public `[FIRApp validateAppID:]` here, then the tests would not change at all (because only implementation details updated).